packages/story-editor/src/app/media/utils/useFFmpeg.js

```diff
@@ -282,6 +282,65 @@ function useFFmpeg() {
     [getFFmpegInstance]
   );
 
+  /**
+   * Segment a video using FFmpeg.
+   *
+   * @param {File} file Original video file object.
+   * @param {string} segmentTime number of secs to split the video into.
+   * @return {Promise<File[]>} Segmented video files .
+   */
+  const segmentVideo = useCallback(
+    async (file, segmentTime) => {
+      let ffmpeg;
+      try {
+        ffmpeg = await getFFmpegInstance(file);
+        const type = file?.type || MEDIA_TRANSCODED_MIME_TYPE;
+        const ext = getExtensionFromMimeType(type);
+        const outputFileName = getFileBasename(file) + '_%03d.' + ext;
+        await ffmpeg.run(
```

Comment:
Should we optimize this video at the same time?

Reply:
Related Slack discussion: https://xwp.slack.com/archives/C013J2EGH8R/p1663840238602189?thread_ts=1663837781.331869&cid=C013J2EGH8R

Since the same can be asked for trimming and cropping as well, I'd suggest creating a new ticket to discuss this.